js/views/SurveyListPage.js

```diff
@@ -145,25 +145,25 @@ const SurveyListPage = React.createClass ({
   render() {
     if (this.state.isLoading) {
       return (<Loading/>)
-    } else if(this.state.dataSource.getRowCount() > 0) {
```

Comment:
That code hides the bar if you click an empty filter section, making you unable to go back to the populated ones.